Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -794,10 +795,19 @@ void BlockFormattingContext::layout_viewport(LayoutMode layout_mode, AvailableSp
     auto& viewport = verify_cast<Layout::Viewport>(root());
     auto& viewport_state = m_state.get_mutable(viewport);
 
-    if (root().children_are_inline())
-        layout_inline_children(root(), layout_mode, available_space);
-    else
-        layout_block_level_children(root(), layout_mode, available_space);
+    // NOTE: If we are laying out a standalone SVG document, we give it some special treatment:
+    //       The root <svg> container gets the same size as the viewport,
+    //       and we call directly into the SVG layout code from here.
+    if (root().first_child() && root().first_child()->is_svg_svg_box()) {
+        auto const& svg_root = verify_cast<SVGSVGBox>(*root().first_child());
+        auto svg_formatting_context = create_independent_formatting_context_if_needed(m_state, svg_root);
+        svg_formatting_context->run(svg_root, layout_mode, available_space);
+    } else {
+        if (root().children_are_inline())
+            layout_inline_children(root(), layout_mode, available_space);
+        else
+            layout_block_level_children(root(), layout_mode, available_space);
+    }
```

Comment:
A layout test for this block would be a good thing to have (just duplicate `svg-as-image` with overridden metrics.)

Reply:
Since we don't currently dump out the SVG-as-image layout tree, we can't see the effects of this directly.
This is a good point though, we should make those trees visible, both DOM and layout, so we can test them. Will add this shortly :)